twisted/twisted
# [9377] Don't remove trailing spaces from SSH version string comment.

Scope
====

See https://twistedmatrix.com/trac/ticket/9377 for a description of the bug.

Conch should keep the SSH version string as provided by the peer, even if it has trailing spaces.


Changes
=======

Update the code to only remove LF character

As a drive by, I have updated the comments to inform that Twisted also support simple Unix new line separate and  wrote a test to explicitly cover this behavior

How to test / What to check
======================

For backward compatibility, I left the non-RFC behaviour in place and just documented.
Let me know if you think that Twisted conch should be more strict and reject peers which only use unix new line... this will be done in a separate ticket as we will need to update some tests.

I wrote unit tests and I think that they are good enough.

To manual test this in the wild you need a Bitvise server which advertise the version with spaces

See https://github.com/mscdex/ssh2/issues/488#issuecomment-329085866 for more details.
